lib/puppet/provider/user/useradd.rb

```diff
@@ -152,6 +152,46 @@ def groups=(value)
     set(:groups, value)
   end
 
+  def password=(value)
+    user = @resource[:name]
+
+    begin
+      # Puppet execute does not support strings as input, only files.
+      # The password is expected to be in an encrypted format given -e is specified:
+      tempfile = nil
+      tempfile = Tempfile.new("puppet_#{user}_pw", :encoding => Encoding::ASCII)
+      tempfile << "#{user}:#{value}\n"
+      tempfile.close()
```

Comment:
I think we want to flush instead of close here. If we close then someone could overwrite the file before we call `chpasswd` (assuming they could overcome the tempfile permissions 0600)